lib/core/bytes.nit

```diff
@@ -328,42 +328,64 @@ class Bytes
 		return new FlatString.full(ns, elen, 0, elen)
 	end
 
-	# Interprets `self` as a big-endian positive integer.
+	# Interprets `self` as a big-endian integer (unsigned by default)
 	#
 	# ~~~
 	# var b = "0102".hexdigest_to_bytes
 	# assert b.to_i == 258
+	#
+	# assert   "01".hexdigest_to_bytes.to_i == 1
+	# assert   "FF".hexdigest_to_bytes.to_i == 255
+	# assert "0000".hexdigest_to_bytes.to_i == 0
 	# ~~~
 	#
-	# Nul bytes on the left are trimmed.
-	# 0 is returned for an empty Bytes object.
+	# If `self.is_empty`, 0 is returned.
 	#
 	# ~~~
-	# assert "01".hexdigest_to_bytes.to_i == 1
-	# assert "0001".hexdigest_to_bytes.to_i == 1
-	#
-	# assert "0000".hexdigest_to_bytes.to_i == 0
-	# assert "00".hexdigest_to_bytes.to_i == 0
 	# assert "".hexdigest_to_bytes.to_i == 0
 	# ~~~
 	#
+	# If `signed == true`, the bytes are read as a signed integer.
+	#
+	# ~~~
+	# assert     "01".hexdigest_to_bytes.to_i(true) ==      1
+	# assert     "FF".hexdigest_to_bytes.to_i(true) ==     -1
+	# assert     "E0".hexdigest_to_bytes.to_i(true) ==    -32
+	# assert   "FE00".hexdigest_to_bytes.to_i(true) ==   -512
+	# assert "FEFEFE".hexdigest_to_bytes.to_i(true) == -65794
+	# ~~~
```

Comment:
you should indicate that for signed integers, the size in bytes (not bits) is inferred from the number of pairs of hex digits.

~~~nit
assert      "F".hexdigest_to_bytes.to_i(true) ==      15
assert     "0F".hexdigest_to_bytes.to_i(true) ==      15
assert     "FF".hexdigest_to_bytes.to_i(true) ==      -1
assert    "0FF".hexdigest_to_bytes.to_i(true) ==     255
assert    "FFF".hexdigest_to_bytes.to_i(true) ==    4095
assert   "0FFF".hexdigest_to_bytes.to_i(true) ==    4095
assert   "FFFF".hexdigest_to_bytes.to_i(true) ==      -1
~~~

￼

Reply:
Add it to `hexdigest_to_bytes` maybe? In `Bytes::to_i` we have the size directly as `length`.

Or I could update the examples to use the literal byte strings, it may be more clear.

Reply:
I don't know. maybe you could just be more explicit about the that only the first byte determines the sign if `true` is given.

You're right that my examples might be more confusing that needed. they are more about`hexdigest_to_bytes` than `to_i` in fact.